lib/datastore/transaction.js

```diff
@@ -119,12 +123,10 @@ Transaction.prototype.begin = function(callback) {
  * @param {function} callback - The callback function.
  *
  * @example
- * transaction.begin(function(err) {
- *   transaction.rollback(function(err) {
- *     if (err) {
- *       // Transaction could not be rolled back.
- *     }
- *   });
+ * transaction.rollback(function(err) {
```

Comment:
This is actually a bit confusing, because a rollback can only be done on a transaction that has been committed (I think?), but that's a problem:

```js
dataset.runInTransaction(function(transaction, _commit) {
  
}, function(err) {
  // no `transaction` here to call `rollback` on.
})
```

Also, is there a use case for a user wanting to rollback a transaction that they didn't just create? 

Sorry all this transaction stuff is getting lumped into this PR.